### PR TITLE
[AttestationVerifier] Do not return error when ExtractSKIDFromX509Cert fails

### LINKED ISSUE #22331: [AttestationVerifier] We should not return error when ExtractSKIDFromX509Cert fails
#### Problem
We cannot guarantee that there are no other der files under the PAA trust store path. For example, if the Certification Declaration file is in the PAA trust store path and it is read before the PAA cert file, the GetProductAttestationAuthorityCert will return ERROR before it gets the right PAA cert.

 #### Proposed Solution
Skip the der file or do not read the der file if it does not have a SKID

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
Fixes #22331
We should not return error when `ExtractSKIDFromX509Cert` fails in `GetProductAttestationAuthorityCert` function of `FileAttestationTrustStore`

#### Change overview
Change it to `continue` now.

#### Testing
Pass CI

### CODE REVIEW

File: src/credentials/attestation_verifier/FileAttestationTrustStore.cpp [hunk 1]
```diff
@@ -104,9 +104,10 @@ CHIP_ERROR FileAttestationTrustStore::GetProductAttestationAuthorityCert(const B
     {
         uint8_t skidBuf[Crypto::kSubjectKeyIdentifierLength] = { 0 };
         MutableByteSpan candidateSkidSpan{ skidBuf };
-        VerifyOrReturnError(CHIP_NO_ERROR ==
-                                Crypto::ExtractSKIDFromX509Cert(ByteSpan{ candidate.data(), candidate.size() }, candidateSkidSpan),
-                            CHIP_ERROR_INTERNAL);
+        if (CHIP_NO_ERROR != Crypto::ExtractSKIDFromX509Cert(ByteSpan{ candidate.data(), candidate.size() }, candidateSkidSpan))
```
Comment (tcarmelveilleux) [inline src/credentials/attestation_verifier/FileAttestationTrustStore.cpp:107]: No item in the trust store should fail to have a SKID. Why do you see an error happen here?

**Review by tcarmelveilleux — CHANGES_REQUESTED** [review]
- [x] All PAAs should have subject key ID by spec. None should be in the trust store location without a SKID. Please explain why this change is needed
- [x] Need an issue number justifying why needs to be fixed

File: src/credentials/attestation_verifier/FileAttestationTrustStore.cpp (hunk 1 above)
Comment (wqx6) [reply src/credentials/attestation_verifier/FileAttestationTrustStore.cpp:107]: We cannot guarantee that there are no other der files under the PAA trust store path. In my test, if the Certification Declaration file is in the PAA trust store path and it is read before the PAA cert file, the `GetProductAttestationAuthorityCert` will return ERROR before it gets the right PAA cert.

I think we should skip the der file if it does not have a SKID.

Comment (wqx6) [reply src/credentials/attestation_verifier/FileAttestationTrustStore.cpp:107]: I have created an issue #22331

**Review by tcarmelveilleux — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

**Comment by andy31415** [comment]
Approved for 1.0: interoperability bindings, useful for general API integrators.

**Pull request merged**